hubblestack/hec/dq.py

```diff
@@ -0,0 +1,127 @@
+
+import time
+from diskcache import Deque
+from sqlite3 import DatabaseError as SQLiteDBError
+import shutil, os
+import glob
+
+# fakelogging goes to a file iff specified in environment
+# and otherwise does nothing (for debugging)
+import fakelogging as logging
+log = logging.getLogger('dq')
+
+class DataFormatError(ValueError):
+    pass
+
+class DiskQueue(object):
+    last_put = last_get = 0
+
+    def __init__(self, directory, max_items=None, max_size=None, restrict_to=None):
+        self.directory = directory
+        self.max_items = max_items
+        self.max_size  = max_size
+        self.restrict_to = restrict_to
+
+    @property
+    def q(self):
+        ''' Open the cache and return the Deque object.  It's desirable for
+        this instance to go out of scope frequently to keep the journal size
+        nominal.
+
+        If the SQLite3 file that backs the deck gets corrupted or if there's
+        SQLite3 version mismatches (etc) this property should nuke the deck and
+        quietly start from scratch.
+        '''
+
+        try:
+            return Deque(directory=self.directory)
+        except SQLiteDBError as e:
+            # The notion is: if we can't open the database because it's
+            # corrupted or the wrong sqlite version or whatever; just give up
+            # and consider those msgs lost.
+            log.error("problem opening the diskqueue at %s: %s", self.directory, e)
+            log.error("deleting the diskqueue")
+            cache_file = os.path.join(self.directory, 'cache.db')
+            if os.path.isfile(cache_file):
+                for f in glob.glob(cache_file + '*'):
+                    os.unlink(f)
+            # If it still doesn't work, crash in the previous way
+            return Deque(directory=self.directory)
```

Comment:
I think you can just `raise` and it will raise the exception from this context. Should be more performant (and more precise)

Reply:
It's a comment bug, not a performance issue or a raise style problem...

We unlinked the old sqlite3 file assuming some sort of journal or version mismatch or something, then we re-try the write but without catching it.

If there's any bug here (aside from the comment, which I'll re-phrase), it's probably that this could take the whole hubble-daemon down with it.

Reply:
ohhhh I totally misread this. Sorry.